drivers/src/sha384acc.rs

```diff
@@ -36,22 +43,46 @@ impl Sha384Acc {
     ///
     /// # Arguments
     ///
-    /// * None
+    /// * assumed_lock_state - The assumed lock state of the SHA384 Accelerator.
+    /// Note: Callers should pass assumed_lock_state=ShaAccLockState::NotAcquired
+    ///  unless they are the first caller to the peripheral after a cold/warm boot.
     ///
     /// # Returns
     ///
-    /// * `Sha384AccOp` - On, success, an object representing the SHA384 accelerator operation.
-    /// * 'None' - On failure to acquire the SHA384 Accelerator lock.
-    pub fn try_start_operation(&mut self) -> Option<Sha384AccOp> {
+    /// * On success, either an object representing the SHA384 accelerator operation or
+    /// 'None' if unable to acquire the SHA384 Accelerator lock.
+    /// On failure, an error code.
+    ///
+    pub fn try_start_operation(
+        &mut self,
+        assumed_lock_state: ShaAccLockState,
+    ) -> CaliptraResult<Option<Sha384AccOp>> {
         let sha_acc = self.sha512_acc.regs();
 
-        if sha_acc.lock().read().lock() && sha_acc.status().read().soc_has_lock() {
-            None
-        } else {
-            // We acquired the lock, or we already have the lock (such as at startup)
-            Some(Sha384AccOp {
-                sha512_acc: &mut self.sha512_acc,
-            })
+        match assumed_lock_state {
+            ShaAccLockState::NotAcquired => {
+                if sha_acc.lock().read().lock() {
+                    // Either SOC has the lock (correct state),
+                    // or the uC has the lock but the caller doesn't realize it (bug).
+                    Ok(None)
+                } else {
+                    // The uC acquired the lock just now.
+                    Ok(Some(Sha384AccOp {
+                        sha512_acc: &mut self.sha512_acc,
+                    }))
+                }
+            }
+            ShaAccLockState::AssumedLockState => {
+                if sha_acc.lock().read().lock() {
+                    // The uC had the lock, and the caller is assuming (correctly) that the uC already has it.
```

Comment:
> the caller is assuming (correctly) that the uC already has it.

From the context of this code, we don't know that, as we're not verifying it here. If the SoC had the lock, we would still go into this branch.

Reply:
Maybe this could read:

"Somebody has the lock, and the caller is assuming that it's the uC".